aws-ssm-document/src/main/java/com/amazonaws/ssm/document/tags/TagReader.java

```diff
@@ -0,0 +1,46 @@
+package com.amazonaws.ssm.document.tags;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import software.amazon.awssdk.services.ssm.SsmClient;
+import software.amazon.awssdk.services.ssm.model.Tag;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+
+@RequiredArgsConstructor
+public class TagReader {
+    private static TagReader INSTANCE;
+
+    @NonNull
+    private final TagClient tagClient;
+
+    public static TagReader getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new TagReader(TagClient.getInstance());
+        }
+
+        return INSTANCE;
+    }
+
+    public Map<String, String> getDocumentTags(@NonNull final String documentName,
+                                                     @NonNull final SsmClient ssmClient,
+                                                     @NonNull final AmazonWebServicesClientProxy proxy) {
+        final List<Tag> documentTags = tagClient.listTags(documentName, ssmClient, proxy);
+
+        return translateTags(documentTags);
+    }
+
+    private Map<String, String> translateTags(final List<Tag> tags) {
+        final ImmutableMap.Builder<String, String> tagBuilder = ImmutableMap.builder();
+
+        tags.forEach(tag -> tagBuilder.put(tag.key(), tag.value()));
```

Comment:
can there be tags with duplicate keys? 

nit: consider using `Collectors.toMap` to convert list of tags to map of tag key/value?

Reply:
no I don't there can't be duplicate keys. We're just returning the tags that we got from API response, so we wouldn't have to worry about it.

Reply:
updated to use ImmutableMap collection lambda